### ISSUE TITLE
BLS signature verification fails on mirror nodes

### ISSUE THREAD

**Issue by drgorb** [hashgraph/hedera-smart-contracts#383]
### Description

when calling the `verifyDefaultMessage` view function on the contract at `0.0.1183384` on the testnet, the call succeeds over the SDK calling the consensus node but fails over a mirror node.

### Steps to reproduce

call the function over the SDK:

```
import { Client, ContractByteCodeQuery, ContractCallQuery, Hbar, HbarUnit } from '@hashgraph/sdk';
import { accountId, derPrivateKey } from "./account";

async function main() {
  const client = Client.forTestnet();
  client.setOperator(accountId, derPrivateKey);
  client.setMirrorNetwork('testnet.mirrornode.hedera.com:443')

  const contractId = "0.0.1183384";
  const scRead2 = new ContractCallQuery()
      .setContractId(contractId)
      .setGas(1_000_000)
      .setFunction('verifyDefaultMessage')
  let cost = await scRead2.getCost(client);
  cost = new Hbar(cost.to(HbarUnit.Tinybar).multipliedBy(1).dividedToIntegerBy(1), HbarUnit.Tinybar);
  console.log(`- Cost to query the bytecode for the contract: ${cost.to(HbarUnit.Hbar).toString()}`);
  const scRead2Tx = await scRead2.setQueryPayment(cost).execute(client);
  const scRead2ReturnValue = scRead2Tx.getBool();
  console.log(`- The verifyDefaultMessage() returned: ${scRead2ReturnValue}`);
  return scRead2ReturnValue ? "Success" : "Failure";
}

main().then(res => {
  console.log(res);
  process.exit(0)
}).catch(err => {
  console.error(err);
  process.exit(1);
});
```
this works but it costs HBAR to execute

call the same function over a mirror node:

```
curl --location 'https://testnet.mirrornode.hedera.com/api/v1/contracts/call' \
--header 'Content-Type: application/json' \
--data '{
  "block": "latest",
  "data": "0x90fb5717",
  "estimate": true,
  "from": "0x688b16d2f18c2043ee1d8ba4a0310323fef29f97",
  "gas": 2000000,
  "gasPrice": 100000000,
  "to": "0x0000000000000000000000000000000000120e98",
  "value": 0
}'
```
this call is free but the function reverts

### Additional context

the contract and associated code to reproduce the bug can be found in this repo: https://github.com/drgorb/bn254_sign_verify

@Pathorn was able to reproduce the error as he said on discord: https://discord.com/channels/373889138199494658/909532351388864542/1149667961594978344

### Hedera network

testnet

### Version

v2.33.0

### Operating system

None

**Comment by drgorb** [comment]
I think the issues comes from this call:

```
        assembly {
            callSuccess := staticcall(
                precompileGasCost,
                8,
                input,
                384,
                out,
                0x20
            )
        }
```

which can be found in [the BLS library](https://github.com/drgorb/bn254_sign_verify/blob/5b648c5692bf5a0e73cae4b3f5456913f9e38999/contracts/BLS.sol#L56C10-L56C10)

this calls the precompile at `0x08` ecPairing (EIP-196 and EIP-197)

**Comment by IvanKavaldzhiev** [comment]
Hi,

we implemented a fix for the reported problem, described in this [issue](https://github.com/hashgraph/hedera-mirror-node/issues/6861). The fix should be available in 0.89.0 mirror-node version.

**Comment by drgorb** [comment]
I tested, it works now
